vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/pieces/select_from_paste_modal.tsx

```diff
@@ -0,0 +1,385 @@
+    return userValues.filter((val) => {
+      const nearests = ([] as string[]).concat(...Object.values((valueMatches as valueMatches)[val]));
+      return !nearests.some(nearest => selectedMatches.includes(nearest));
+    });
+  }, [userValues, valueMatches, selectedMatches]);
+
+  // onClick for matches
+  function addOrRemove(match: string) {
+    if (selectedMatches.includes(match)) {
+      setSelectedMatches(selectedMatches.filter(v => v!=match));
+    } else {
+      setSelectedMatches(selectedMatches.concat([match]));
+    };
+  };
+
+  // onClick for Search button
+  function parseAndMatchText() {
+    setBusy(true);
+
+    // Parse Input Text to discrete search elements
+    const parsedValues = userText.split(/[\s\n,;]+/).filter((val) => !['',' '].includes(val));
+    if (parsedValues.length<1) {
+      setBusy(false);
+      return;
+    };
+    setUserValues(parsedValues);
+
+    // Perform fuzzy search per each options set
+    const setMatches: {[OptionSet: string]: ({
+      userValue: string,
+      useMatch: boolean,
+      bestMatch: string,
+      topMatches: string[]
+    } | {
+      userValue: string,
+      useMatch: false,
+      bestMatch: null,
+      topMatches: string[]
+    })[]} = {};
+    for (const set of Object.keys(options)) {
+      const fuse = new Fuse(options[set], fuseOptions);
+
+      setMatches[set] = parsedValues.map((val: string) => {
+        let topMatches = fuse.search(val).slice(0,20);
+        // Case: No matches picked at all
+        if (topMatches.length<1) {
+          return {
+            userValue: val,
+            useMatch: false,
+            bestMatch: null,
+            topMatches: [] as string[]
+          };
+        };
+        // Case: No near-exact match.
+        // For genomics data, pull thought-to-be desired top hits forward
+        // Targets:
+        //   - because of CD#s: letters, not numbers, following the query
+        //   - because of how scData tools make markers unique: '.#' following the query.
+        if (genomicData && topMatches[0].score as number >= 0.001) {
+          const likely: number[] = [];
+          const lesslikely: number[] = [];
+          const valNoSymbol: string = val.replace(/\W/g,'.'); // to protect against symbols in the user input
+          for (let ind = 0; ind < topMatches.length; ind++) {
+            if (topMatches[ind].item.val.search(RegExp(valNoSymbol+'[a-z]|'+valNoSymbol+'\\.\\d', 'i')) > -1) {
+              likely.push(ind);
+            } else {
+              lesslikely.push(ind);
+            };
+          };
+          topMatches = likely.concat(lesslikely).map(i => topMatches[i]);
+        };
+
+        // Return (up to) top 5 matches
+        return {
+          userValue: val,
+          useMatch: topMatches[0].score as number < 0.001,
+          bestMatch: topMatches[0].item.val as string,
+          topMatches: topMatches.map((v: any) => v.item.val as string).slice(0,5) as string[]
```

Comment:
Make configurable the max number of matches returned?

Reply:
Added a slider for this, and had to muck with spacing to make it look right... so in the process also messed with MUIs Grid `xs` and `md` tags to improve space utilization on smaller screens =)

How do you like the look of the slider and heading?
![bulk-add-withSlider](https://github.com/mountetna/monoetna/assets/38870347/ae80023e-95f4-47d4-aedd-33fb650742e1)

:+1: and I'll take it as I should still go ahead and merge!